beast/tools/make_libfile.py

```diff
@@ -339,5 +646,66 @@ def make_libfile():
     hf.close()
 
 
+def make_vega_libfile():
+
+    # Read an updated filters.hd5 lib
+    __default_filtlist__ = __ROOT__ + "/filters.hd5"
+    filtlist = QTable.read(__default_filtlist__, path="content")
+    filters = [cfilt.decode("UTF-8") for cfilt in filtlist["TABLENAME"].data]
+
+    # filenames for vega info
+    __default_vega__ = __ROOT__ + "/vega.hd5"
+    __default_vega_old__ = __ROOT__ + "/vega_old.hd5"
+
+    # rename the current file so we can write a new version
+    os.rename(__default_vega__, __default_vega_old__)
+
+    # Get a spectrum from the need-to-be updated vega lib
+    vega_old = QTable.read(__default_vega_old__, path="spectrum")
+    vl = vega_old["WAVELENGTH"].data
+    vf = vega_old["FLUX"].data
+
+    # Write out an updated vega.hd5 file
+    vega = h5py.File(__default_vega__, "w")
+
+    vega.create_dataset("spectrum", data=vega_old)
+    flist = phot.load_filters(
+        filters, interp=True, lamb=vl, filterLib=__default_filtlist__
+    )
+
+    fname = []
+    cwave = []
+    lum = []
+    mag = []
+
+    for cfilt in flist:
+        fname.append(cfilt.name)
+        cwave.append(cfilt.cl)
+        flux = cfilt.getFlux(vl, vf)
+        lum.append(flux)
+        mag.append(-2.5 * np.log10(flux))
+
+    contents = np.array(
+        list(
+            zip(
+                fname,
+                cwave,
+                lum,
+                mag,
+            )
+        ),
+        dtype=[
+            ("FNAME", "S30"),
+            ("CWAVE", "<f8"),
+            ("LUM", "<f8"),
+            ("MAG", "<f8"),
+        ],
+    )
+
+    vega.create_dataset("sed", data=contents)
+    vega.close()
```

Comment:
Would it be good if we close vega_old and filters tables as well?  Or it's not important.

Reply:
Actually, do not need to close those as I changed to using the astropy.table QTable function to read them.  They are not kept open the same way that h5py does.   Might be able to change the vega writing to use the same interface, but likely would need to do more testing.  Trying to move to just using astropy.table functions to avoid such "low" level interface issues.